File: python/clase-02/final/main.py
"""Conversación multi-turn: el cliente guarda y reenvía el historial."""

import os
from anthropic import Anthropic

MODEL = "claude-3-5-sonnet-latest"


def main() -> None:
    api_key = os.getenv("ANTHROPIC_API_KEY")
    if not api_key:
        raise RuntimeError("Define ANTHROPIC_API_KEY.")

    client = Anthropic(api_key=api_key)
    messages = [
        {"role": "user", "content": "Mi proyecto será un chatbot para recetas."},
        {"role": "assistant", "content": "Perfecto. Puedo ayudarte con ingredientes y pasos."},
        {"role": "user", "content": "Recuérdame cuál era mi proyecto y sugiere el primer feature."},
    ]

    response = client.messages.create(model=MODEL, max_tokens=400, messages=messages)
    for block in response.content:
        if block.type == "text":
            print(block.text)


if __name__ == "__main__":
    main()

File: typescript/clase-02/final/main.ts
import Anthropic from "@anthropic-ai/sdk";

export {};

const MODEL = "claude-sonnet-4-6";

const apiKey = process.env.ANTHROPIC_API_KEY;
if (!apiKey) throw new Error("Define ANTHROPIC_API_KEY.");

const client = new Anthropic({ apiKey });
const response = await client.messages.create({
  model: MODEL,
  max_tokens: 400,
  messages: [
    { role: "user", content: "Mi proyecto será un chatbot para recetas." },
    { role: "assistant", content: "Perfecto. Puedo ayudarte con ingredientes y pasos." },
    { role: "user", content: "Recuérdame cuál era mi proyecto y sugiere el primer feature." },
  ],
});

for (const block of response.content) {
  if (block.type === "text") console.log(block.text);
}
